cli/main.rs

```diff
@@ -1076,9 +1078,20 @@ async fn run_with_watch(flags: Flags, script: String) -> Result<i32, AnyError> {
         flags.compat,
       );
 
-      executor.execute(&main_module).await?;
+      let result = executor.execute(&main_module).await;
+      let mut exit_code = executor.worker.get_exit_code();
 
-      Ok(())
+      if let Err(err) = result {
+        // `Deno.exit()` was called
+        // TODO(bartlomieju): this is brittle
+        if err.to_string() != "Uncaught Error: execution terminated" {
+          let msg = format!("{}: {}", colors::red_bold("error"), err);
+          eprintln!("{}", msg);
+          exit_code = 1;
+        }
+      }
```

Comment:
It seems like this could stand to have a more deterministic flag on the error. Can you catch specific kinds of errors? such as `let TerminatedErr(err) = result` ? 

Also this looks very similar to the `error_handler2 ` function in file_watcher.rs, could it not be shared?

Reply:
This is still WIP, I need to merge a few PRs before this one can be closed. I will add necessary infra in other PR.